lib/vector-core/src/event/array.rs

```diff
@@ -106,7 +110,7 @@ impl EventContainer for MetricArray {
 }
 
 /// An array of one of the `Event` variants exclusively.
-#[derive(Debug, PartialEq)]
+#[derive(Clone, Debug, PartialEq)]
 pub enum EventArray {
```

Comment:
Nit: I know we talked about this earlier today too, but, seeing it now, maybe these types could just be called the plural? I.e.:

```rust
pub enum Events {
  Logs(Logs)
  Metrics(Metrics)
}
```

Or would that be confusing?

The use of `Array` is making me think the backing type is going to be a literal array though I know "array" has a more general definition.

Naming is hard.

Either way, nothing that needs to change in this PR.

Reply:
I think making them just plural is a little more visually subtle than I'd like. FWIW I did experiment with making it a literal `Box<[LogEvent]>` etc, but there were a couple of pain points. Once the events are in one of these types, they aren't generally modified except to stream them all out and move them into another (other than in tests), so "array" isn't necessarily far from the truth.

Reply:
`EventBatch`? or `EventCollection`?

Reply:
The term "batch" is IMO by far the most appropriate, but we were shying away from using the since vector already uses that term for something else (and they are batches of events no less).

Reply:
Alternatively, `EventGroup` could be considered as a more "neutral" variant between `EventArray` and `EventBatch`.

Reply:
FWIW I like `EventArray`.